shell/src/main/java/alluxio/cli/fsadmin/FileSystemAdminShell.java

```diff
@@ -79,15 +83,24 @@ protected String getShellName() {
   protected Map<String, Command> loadCommands() {
     ClientContext ctx = ClientContext.create(mConfiguration);
     MasterClientContext masterConfig = MasterClientContext.newBuilder(ctx).build();
-    Context context = new Context(
+    mContext = new Context(
         new RetryHandlingFileSystemMasterClient(masterConfig),
         new RetryHandlingBlockMasterClient(masterConfig),
         new RetryHandlingMetaMasterClient(masterConfig),
         new RetryHandlingMetaMasterConfigClient(masterConfig),
         System.out
     );
     return CommandUtils.loadCommands(FileSystemAdminShell.class.getPackage().getName(),
-        new Class[] {Context.class, AlluxioConfiguration.class}, new Object[] {context,
+        new Class[] {Context.class, AlluxioConfiguration.class}, new Object[] {mContext,
             mConfiguration});
   }
+
+  @Override
+  public void close() throws IOException {
+    super.close();
+    mContext.getBlockClient().close();
```

Comment:
why not make `Context` closable and using `Closer` inside `Context.close()` to make sure we try the best to close all resources?